library/tiqr/Tiqr/StateStorage/Pdo.php

```diff
@@ -25,17 +25,39 @@
 
 
 class Tiqr_StateStorage_Pdo extends Tiqr_StateStorage_Abstract
-{    
-    private $handle = null;
+{
+    /**
+     * @var PDO
+     */
+    protected $handle;
+
+    /**
+     * @var string
+     */
     private $tablename;
-        
+
+    /**
+     * @var int
+     */
+    private $cleanupProbability;
+
```

Comment:
My suggestion would be to specify cleanupProbability as a float with allowed values 0 >= cleanupProbability >= 1. This is also the way probability is defined in statistics:
0: Cleanup will never run
0.2: Cleanup will run in 20% of the calls to setValue
1: Cleanup will always run

Using "1000" limits the resolution and the current implementation and binds cleanupProbability to this choice making it much harder to change this later, so instead I'd do (pseudocode):

```
if ( $cleanupProbability > (rand() / getrandmax()) ) {
   cleanup()
}
```

Reply:
Yup! that makes way more sense.